[DO NOT MERGE] Update roadmap

andysellick opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

**DO NOT MERGE** awaiting final copy

## What / why

Update the roadmap page. Mostly content changes, with a few layout changes, specifically:

- moves the visit counter to the bottom of the page
- adds three links to other pages with images at the top of the page
- adds sharing links at the bottom

## Visual changes

Before | After
------ | ------
![screencapture-gov-uk-roadmap-2021-04-07-11_34_58](https://user-images.githubusercontent.com/861310/113853239-57916600-9795-11eb-8aba-ae83e3bf95a1.png) | ![screencapture-localhost-3005-roadmap-2021-04-07-11_35_35](https://user-images.githubusercontent.com/861310/113853296-6a0b9f80-9795-11eb-802b-fd1e44349c91.png)

andysellick commented:
When you review, please also check that the sharing links at the bottom of the page work as expected.

danacotoran commented:
@andysellick if this is to be shared on social media I think we should use metadata to improve the way it would be previewed.
As it is currently, if shared on Facebook it would look a little something like this:
<img width="524" alt="Screenshot 2021-04-20 at 10 15 43" src="https://user-images.githubusercontent.com/7116819/115372183-8c081780-a1c2-11eb-9ac6-651067817f44.png">
Which is fine, but perhaps a little bit generic? I wonder if we should be more deliberate with it.

If shared on Twitter it would simply appear as a link, as none of the metadata to generate a twitter share card is there. 

This definitely needs some custom Open Graph meta tags (my bad for missing this in the original implementation of this page)!

andysellick commented:
@danacotoran I've updated the copy today and added the new links for sharing. I don't have either of these accounts (twitter, linkedin) so can you test the links work properly please?

> This definitely needs some custom Open Graph meta tags (my bad for missing this in the original implementation of this page)!

Have added some of these as well.

danacotoran commented:
Checked the social links and this looks good to me now, but it might be good to get a content designer to check the copy updates properly as I'm not the best at spotting issues with that. 
One thing that stood out to me visually was that on some breakpoints the text in only one of the  "Programme updates" columns wraps onto the next line which does not look great but also not the end of the world if it's not an easy thing to fix.
<img width="575" alt="Screenshot 2021-05-14 at 11 49 46" src="https://user-images.githubusercontent.com/7116819/118260718-8d1f2280-b4aa-11eb-8c41-043f37b8077f.png">

keithiopia commented on config/locales/en.yml:
```diff
@@ -102,126 +102,136 @@ en:
-          - text: Improve the way we manage subdomains of GOV.UK for the rest of government
-          - text: Ongoing work on cookies
-          - text: Ongoing work on accessibility
+          - text: Continue work to use <a href="https://schema.org/" class="govuk-link">schema.org</a> standards on GOV.UK
```
This should be:
```suggestion
          - text: Continue work to use <a href="https://dataingovernment.blog.gov.uk/2020/04/30/using-new-updates-to-schema-org-to-assist-our-response-to-the-coronavirus-pandemic/?utm_medium=referral%20&utm_source=roadmap_gov&utm_campaign=roadmap_update" class="govuk-link">schema.org standards on GOV.UK</a>
```

andysellick replied:
Thanks Keith, have updated.

danacotoran commented on config/locales/en.yml:
```diff
@@ -111,179 +111,139 @@ en:
+      items:
+      - href: "https://gds.blog.gov.uk/2021/03/29/whats-next-for-gov-uk-in-2021-to-2022/"
+        image_src: "roadmap/updates-goals.jpg"
+        image_alt: "alt text"
```
These images are decorative. Can we please leave alt text blank so that screen readers skip over them?

danacotoran commented on config/locales/en.yml:
```diff
@@ -111,179 +111,139 @@ en:
+          - text: Experiment with audience analysis across sections of GOV.UK to understand page usage vs reading time to support product and content iteration
+          - text: Simplify how feedback is analysed and gathered, so we can better measure our impact and identify issues
+      - heading: Provide a personalised and proactive service
+        paragraph: Develop sitewide account features that offer users relevant information and more helpful notifications
```
We spell `site-wide` without the hyphen in some places, and with the hyphen in other places, I would think it should be consistent throughout the page @keithiopia?

andysellick replied:
I've gone with `site-wide`.

keithiopia replied:
Grammatically speaking, they both work. Good choice.

danacotoran commented on app/views/roadmap/index.html.erb:
```diff
@@ -9,63 +13,42 @@
         <div class="govuk-grid-column-two-thirds">
           <%= render "govuk_publishing_components/components/heading", {
-            text: t('roadmap.numbers_section.heading_large'),
+            text: "Programme updates",
```
Should this text live in the locale file?

andysellick replied:
Yes. In fact it's already in there. Odd. Corrected.

danacotoran reviewed (commented):
I made a few observations, but other than that I think it is looking good 👍

danacotoran commented on app/views/roadmap/index.html.erb:
```diff
@@ -9,63 +13,42 @@
+            <%= render "govuk_publishing_components/components/image_card", {
+              href: item[:href],
+              image_src: asset_path(item[:image_src]),
+              image_alt: item[:image_alt],
```
We don't need to specify `image_alt` because the `image_card` component already [defaults to blank alt text if alt text is not specified](https://github.com/alphagov/govuk_publishing_components/blob/0e23f397270ffc3fc4fd811a84190c308ef9838c/lib/govuk_publishing_components/presenters/image_card_helper.rb#L14).
So I reckon we can remove this line as well as the `image_alt` keys from the locale file.

andysellick replied:
Fair point, removed.

danacotoran reviewed (approved):
Sorry, just had one more comment.
But other than that I think it's good to go for Monday so I am approving 🥇
